### DISCUSSION TITLE
Daily challenge improvements

### DISCUSSION THREAD

**Discussion by sa-shiro** [ppy/osu#29619] [Ideas]
Just lost my daily challenge streak because i couldnt login so i wanna make some suggestions to the system

## Option 1

make daily challenge count on a per-login basis rather than forcefully every single day

## Option 2

Add "osu!tokens" which can be used for:
- skipping current active daily map (this can be very useful for players who need to use relax/nofail because they cant play the map just to keep their streak up)
- exchangeable for a missed day to regain the daily streak (for the current week only)
- users gain 1 "osu!token" each day they login, capped at max 3 in total (5 for osu!supporters)
- _**supporters exclusive**_: can be exchanged for a missed day for the previous week as well
- (extreme case: make osu!tokens purchasable that are separate from login tokens which can be used for the current and last week (or perhaps up to 2 weeks?)

## Additional suggestion

as this feature is still work in progress, it would be nice (and fair i guess?) to reset the daily streak for everyone once osu!lazer officially releases

**Comment by peppy** [comment]
Here's a proposal: let's just let users choose their own streak and call it a day.

Sarcasm aside, I don't see this happening. Just get over it and carry on. The main number on your profile is participation, so breaking a streak should not be a huge deal. Make your mind stronger. I broke a 300 day fitness streak because I was playing games too late. Did I go and request apple change the system? Nah, I started again and now I'm back over 500.

**Comment by sa-shiro** [comment]
While I understand the intention behind the daily challenge streak feature, I believe it could be more inclusive and considerate of players at different skill levels. Forcing players to play every single day might be discouraging, especially for newer or less experienced players who may struggle to maintain a streak for more than a day or two.

It’s important to acknowledge that not all players approach the game with the same level of skill or time commitment. Offering flexibility, such as allowing the use of 'osu!tokens' to skip a day or substitute a difficult map, could make the system more engaging and less punitive. This way, participation would feel rewarding rather than obligatory, ensuring that the feature is enjoyable for a broader range of players.

I do want to mention that the response to the original suggestion came across as a bit dismissive. It's crucial to consider that many players, particularly those less skilled or with limited time, might feel alienated by the current system. A more inclusive design would help ensure that all players, regardless of skill level, can participate and enjoy this feature. I hope future discussions can be more supportive and considerate of the diverse player base that makes up the osu! community.

**Reply by peppy** [reply]
I know where you're coming from but I don't think this is the right direction. We'd sooner allow choosing which difficulty to play from the beatmap than adding some kind of "skip" system.

Besides, playing through a beatmap outside of your comfort zone using NF or Relax gives you insight into where you may be in the future.

**Reply by sa-shiro** [reply]
I appreciate the consideration of allowing players to select the difficulty, as that would indeed make the feature more accessible to a wider range of players. However, I want to clarify that when I mention NF and Relax, I'm speaking on behalf of the broader player base, not just my own experience. It's crucial to consider the diverse needs and skill levels of the entire community.

While choosing difficulties is a positive step, it doesn't fully address the concern that this feature may feel more like a chore than an enjoyable challenge. Players might feel obligated to participate daily, which could detract from the fun. Implementing a system based on a per-login basis, as suggested in Option 1, could effectively address this issue.

This approach would allow players to engage with the game at their own pace without the need for a 'skip' mechanic, which you mentioned as a less favorable direction. By making the feature less about daily obligation and more about consistent engagement, it would enhance the overall experience for everyone.

This feature shouldn't be about competing with others, which in its current state it feels like it. (competing in sense of who has the highest streak)

**Comment by sa-shiro** [comment]
A potential improvement could be to change the "Daily Challenge" to a "Weekly Challenge." For example, players could be required to clear 10 maps of increasing difficulty (or perhaps 8 out of 10) to maintain their weekly streak. This approach would address many of the current issues, such as the pressure to log in every single day. It would also create a more meaningful sense of progression in skill.

This system would give players more flexibility and reduce the anxiety associated with seeing a particularly difficult map. By providing a full week to complete the challenge, players could practice maps they find challenging without feeling rushed. This would encourage more players to engage with higher difficulty maps, possibly using Relax or NoFail mods, as they would have more time to work on their skills. Additionally, having only a weekly counter on the profile would simplify tracking progress and avoid the redundancy of having both daily and weekly streaks displayed.

**Comment by DerHamm3r** [comment]
I'd like to add another suggestion in regards to keeping your streak going if you can't play on a day. Something similar to how chess.com handles their daily streaks could be helpful and it's a solution that doesn't require an entire system change. On chess.com, you can go back 1 day and complete the previous daily puzzle to not lose your streak once you complete the current puzzle. But it only works for going back one day. If you miss 2 or more days, or if you don't go back and complete the previous one, your streak is gone. 

For lazer this could just be you going into solo mode and playing yesterdays daily challenge with the required force mods (if there are any). You wouldn't be able to compete on the previous leaderboards of course, but at least you can keep your streak.

Also one more thing to add: I agree with the point that participation is the main number, but also remember that osu players are obsessed with any number, especially with those that get bigger :)
